packages/@lwc/integration-karma/test/rendering/legacy-scope-tokens/index.spec.js

```diff
@@ -0,0 +1,133 @@
+import { createElement, setFeatureFlagForTest } from 'lwc';
+import Light from 'x/light';
+import Shadow from 'x/shadow';
+
+describe('legacy scope tokens', () => {
+    [false, true].forEach((enableLegacyScopeTokens) => {
+        describe(`enableLegacyScopeTokens=${enableLegacyScopeTokens}`, () => {
+            beforeEach(() => {
+                setFeatureFlagForTest('ENABLE_LEGACY_SCOPE_TOKENS', enableLegacyScopeTokens);
+            });
+
+            afterEach(() => {
+                setFeatureFlagForTest('ENABLE_LEGACY_SCOPE_TOKENS', false);
+            });
+
+            function getAttributes(elm) {
+                return [...elm.attributes]
+                    .map((_) => _.name)
+                    .filter((_) => _ !== 'class')
+                    .sort();
+            }
+
+            function getClasses(elm) {
+                return (
+                    [...elm.classList]
+                        // these are classes we add ourselves for the test
+                        .filter((_) => !['static', 'dynamic', 'manual'].includes(_))
+                        .sort()
+                );
+            }
+
+            function expectTokens(modern, legacy) {
+                return [
+                    ...new Set(
+                        [
+                            process.env.API_VERSION <= 58 ? legacy : modern,
+                            enableLegacyScopeTokens && legacy,
+                        ].filter(Boolean)
+                    ),
+                ].sort();
+            }
+
+            it('light dom', async () => {
+                const elm = createElement('x-light', { is: Light });
+                document.body.appendChild(elm);
+
+                const verify = () => {
+                    const staticDiv = elm.querySelector('.static');
+                    const dynamicDiv = elm.querySelector('.dynamic');
+
+                    expect(getClasses(elm)).toEqual(
+                        expectTokens('lwc-1kadf5igpar-host', 'x-light_light-host')
+                    );
+                    expect(getClasses(staticDiv)).toEqual(
+                        expectTokens('lwc-1kadf5igpar', 'x-light_light')
+                    );
+                    expect(getClasses(dynamicDiv)).toEqual(
+                        expectTokens('lwc-1kadf5igpar', 'x-light_light')
+                    );
+
+                    expect(getAttributes(elm)).toEqual([]);
+                    expect(getAttributes(staticDiv)).toEqual([]);
+                    expect(getAttributes(dynamicDiv)).toEqual([]);
+                };
+
+                await Promise.resolve();
+
+                verify();
+
+                // force re-render
+                elm.rerender = 1;
+                await Promise.resolve();
+
+                verify();
+            });
+
+            if (!process.env.NATIVE_SHADOW) {
```

Comment:
We should test this for native shadow as well, since native shadow gets the classes due to the `*.scoped.css`.